modules/extensions/src/main/java/org/apache/synapse/mediators/bsf/CommonScriptMessageContext.java

```diff
@@ -0,0 +1,848 @@
+            return mc.getEnvelope().toString();
+        }
+    }
+
+    /**
+
+    /**
+     * Saves the payload of this message context as a JSON payload.
+     *
+     * @param jsonPayload Javascript native object to be set as the message body
+     * @throws ScriptException in case of creating a JSON object out of
+     *                         the javascript native object.
+     */
+    public void setPayloadJSON0(Object jsonPayload) throws ScriptException {
+        org.apache.axis2.context.MessageContext messageContext;
+        messageContext = ((Axis2MessageContext) mc).getAxis2MessageContext();
+
+        String jsonString;
+        if (jsonPayload instanceof String) {
+            jsonString = (String) jsonPayload;
+        } else {
+            jsonString = serializeJSON(jsonPayload);
+        }
+        try {
+            JsonUtil.getNewJsonPayload(messageContext, jsonString, true, true);
+        } catch (AxisFault axisFault) {
+            throw new ScriptException(axisFault);
+        }
+        //JsonUtil.setContentType(messageContext);
+        Object jsonObject = scriptEngine.eval('(' + jsonString + ')');
+        setJsonObject(mc, jsonObject);
+    }
+
+    /**
+     * Saves the JavaScript Object to the message context.
+     * @param messageContext
+     * @param jsonObject
+     * @return
+     */
+    @Override
+    public boolean setJsonObject(MessageContext messageContext, Object jsonObject) {
+        if (jsonObject == null) {
+            logger.error("Setting null JSON object.");
+        }
+        messageContext.setProperty(JSON_OBJECT, jsonObject);
+        return true;
+    }
+
+    /**
+     * Saves the JSON String to the message context.
+     * @param messageContext
+     * @param jsonObject
+     * @return
+     */
+    @Override
+    public boolean setJsonText(MessageContext messageContext, Object jsonObject) {
+        if (messageContext == null) {
+            return false;
+        }
+        if (jsonObject == null) {
+            logger.error("Setting null JSON text.");
+        }
+        messageContext.setProperty(JSON_TEXT, jsonObject);
+        return true;
+    }
+
+    /**
+     * Returns the JavaScript Object saved in this message context.
+     * @param messageContext
+     * @return
+     */
+    @Override
+    public Object jsonObject(MessageContext messageContext) {
+        if (messageContext == null) {
+            return null;
+        }
+        Object o = messageContext.getProperty(JSON_OBJECT);
+        if (o == null) {
+            logger.error("JSON object is null.");
+            if (this.scriptEngine == null) {
+                logger.error("Cannot create empty JSON object. ScriptEngine instance not available.");
+                return null;
+            }
+            try {
+                return this.scriptEngine.eval("({})");
+            } catch (ScriptException e) {
+                logger.error("Could not return an empty JSON object. Error>>> " + e.getLocalizedMessage());
```

Comment:
Better to pass the exception to `log.error` as the second argument rather than just printing the localized message